comp/core/autodiscovery/noopimpl/autoconfig.go

```diff
@@ -0,0 +1,96 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package noopautoconfig provides a noop implementation for the autodiscovery component
+package noopautoconfig
+
+import (
+	"context"
+	"time"
+
+	"go.uber.org/fx"
+
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery"
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/providers"
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/scheduler"
+	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/telemetry"
+	checkid "github.com/DataDog/datadog-agent/pkg/collector/check/id"
+	pkgconfigsetup "github.com/DataDog/datadog-agent/pkg/config/setup"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+)
+
+// Module defines the fx options for this component.
+func Module() fxutil.Module {
+	return fxutil.Component(
+		fx.Provide(
+			newAutoConfig,
+		),
+	)
+}
+
+type noopAutoConfig struct{}
+
+func (n *noopAutoConfig) AddConfigProvider(providers.ConfigProvider, bool, time.Duration) {}
```

Comment:
Can you explain the rational of using `noop` here instead of optional?
When a component may or may not be present, in our code base, the common way to handle it is to use optional. One of the reason is that the code that use autodiscovery must work with the default implementation and this noop implementation. Using optional allowed us to catch several issues in the past because of the usage of a wrong usage of the noop implementation. 

Note: Using noop is fine when method needs to return something.